src/legacyConditions.ts

```diff
@@ -0,0 +1,172 @@
+import type {LiquidContext} from './types';
+
+import chalk from 'chalk';
+
+import {tagLine} from './syntax/lexical';
+import {getPreparedLeftContent, removeIndentBlock} from './utils';
+import {evaluate} from './syntax/evaluate';
+
+function getElseProp<B extends keyof Elses>({elses}: {elses: Elses[]}, propName: B, index = 0) {
+    if (!elses.length || index >= elses.length) {
+        return undefined;
+    }
+
+    return elses[index][propName];
+}
+
+type Opts = {
+    ifTag: Tag;
+    vars: Record<string, unknown>;
+    content: string;
+    match: RegExpExecArray;
+    lastIndex: number;
+};
+
+function inlineConditions(this: LiquidContext, {ifTag, vars, content, match, lastIndex}: Opts) {
+    let res = '';
+
+    if (evaluate.call(this, ifTag.condition, vars)) {
+        const ifRawLastIndex = ifTag.startPos + ifTag.ifRaw.length;
+        const contentLastIndex = getElseProp(ifTag, 'startPos') || match.index;
+
+        res = content.substring(ifRawLastIndex, contentLastIndex);
+    } else {
+        ifTag.elses.some(({condition, startPos, raw}, index) => {
+            const isTruthy = !condition || evaluate.call(this, condition, vars);
+
+            if (isTruthy) {
+                const elseRawLastIndex = startPos + raw.length;
+                const contentLastIndex = getElseProp(ifTag, 'startPos', index + 1) || match.index;
+
+                res = content.substring(elseRawLastIndex, contentLastIndex);
+
+                return true;
+            }
+
+            return false;
+        });
+    }
+
+    const preparedLeftContent = getPreparedLeftContent({
+        content,
+        tagStartPos: ifTag.startPos,
+        tagContent: res,
+    });
+
+    let shift = 0;
+    if (
+        res === '' &&
+        preparedLeftContent[preparedLeftContent.length - 1] === '\n' &&
+        content[lastIndex] === '\n'
+    ) {
+        shift = 1;
+    }
+
+    if (res !== '') {
+        if (res[0] === '\n') {
+            res = res.substring(1);
+        }
+
+        res = removeIndentBlock(res);
+
+        if (res[res.length - 1] === '\n') {
+            res = res.slice(0, -1);
+        }
+    }
+
+    const leftPart = preparedLeftContent + res;
+
+    return {
+        result: leftPart + content.substring(lastIndex + shift),
+        idx: leftPart.length,
+    };
+}
+
+type Elses = {startPos: number; raw: string; condition?: string};
+
+type Tag = {
+    isOpen: Boolean;
+    condition: string;
+    startPos: number;
+    ifRaw: string;
+    elses: Elses[];
+};
+
+export = function legacyConditions(
```

Comment:
Since we use ES modules in this project, shouldn’t this be `export default function legacyConditions(...) { ... }` instead of `export =` ?